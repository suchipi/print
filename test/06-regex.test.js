const print = require("../print");

describe("Regular expressions", () => {
	it("prints simple regex", () => {
		expect(print(/a/)).toMatchSnapshot();
		expect(print(/b|c/)).toMatchSnapshot();
		expect(print(/\v/)).toMatchSnapshot();
		expect(print(/a/i)).toMatchSnapshot();
		expect(print(/a/gim)).toMatchSnapshot();
	});

	it("prints hairy regex", async () => {
		const fn = () => [
			/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
			/((\/\*)[^\x00]*?(\*\/))|[\t\n]/gm,
			/\\b(https?:\/\/([-\\w\\.]+)|www\\.|([-\\w\\.]+)\/)\\S+|$/i,
			/^(?:(?:^|\s+)\S+){8}(\s+\S+)/,
			/(<(\w+)\b[^>]*class\s*=\s*("[^">]*pre-line[^">]*"|'[^'>]*pre-line[^'>]*')[^>]*>)(.*?)(<\/\2>)/im,
			/[^\w-_'\u2019]+|['\u2019]{2,}/g,
			/[^\w_'\u2019]+|['\u2019]{2,}/g,
			/"([^\\"]|\\.)*"|'([^\\']|\\.)*'/,
			/((\/\*)[^\x00]*?(\*\/))/gm,
			/(?:^[\x20\t]+)|(?:\n\s*)(?=\n)|\s+$|\n\n/,
			/^[\x20\t]*\n|\n[\x20\t]*(?=\n|$)/gm,
			/^([^/#?]*:?\/\/)?(\/?(?:[^/#?]+\/)*)?([^/#?]+)?(?:\/(?=$))?(\?[^#]*)?(#.*)?$/,
			/^\s*(https?:)?\/\/([^:]+:[^@]+@)?([\w-]+)(\.[\w-]+)*(:\d+)?(\/\S+)?\s*$/,
			/\b(\d*1[1-3]th|\d*0th|(?:(?!11st)\d)*1st|\d*2nd|(?:(?!13rd)\d*)3rd|\d*[4-9]th)\b/,
		];
		const regex = fn();
		const lines = fn
			.toString()
			.replace(/^.+?\[\s*|\s*\]$/gs, "")
			.split(/\r?\n/);
		lines.forEach((line, index) =>
			expect(print(regex[index])).toMatchSnapshot()
		);
	});

	it("displays named properties", () => {
		const value = /abc|xyz/gi;
		value.foo = "Foo";
		expect(print(value)).toMatchSnapshot();
		value.bar = "Bar";
		expect(print(value)).toMatchSnapshot();
	});

	it("identifies subclasses", () => {
		class PCRE extends RegExp {}
		const regex = new PCRE("ABC|XYZ", "gi");
		expect(print(regex)).toMatchSnapshot();
		regex.foo = "Foo";
		expect(print(regex)).toMatchSnapshot();
		regex.exec("ABC");
		expect(print(regex, { all: true })).toMatchSnapshot();
	});
});
