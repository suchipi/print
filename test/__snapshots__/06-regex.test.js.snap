// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regular expressions displays named properties 1`] = `
RegExp {
	/abc|xyz/gi
	
	foo: "Foo"
}
`;

exports[`Regular expressions displays named properties 2`] = `
RegExp {
	/abc|xyz/gi
	
	foo: "Foo"
	bar: "Bar"
}
`;

exports[`Regular expressions identifies subclasses 1`] = `
PCRE {
	/ABC|XYZ/gi
}
`;

exports[`Regular expressions identifies subclasses 2`] = `
PCRE {
	/ABC|XYZ/gi
	
	foo: "Foo"
}
`;

exports[`Regular expressions identifies subclasses 3`] = `
PCRE {
	/ABC|XYZ/gi
	
	lastIndex: 3
	foo: "Foo"
}
`;

exports[`Regular expressions prints hairy regex 1`] = `/^[a-zA-Z0-9.!#$%&'*+/=?^_\`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/`;

exports[`Regular expressions prints simple regex 1`] = `/a/`;

exports[`Regular expressions prints simple regex 2`] = `/b|c/`;

exports[`Regular expressions prints simple regex 3`] = `/\\v/`;

exports[`Regular expressions prints simple regex 4`] = `/a/i`;

exports[`Regular expressions prints simple regex 5`] = `/a/gim`;
